drivers/sdhc/sdhc_litex_litesdcard.c

```diff
@@ -0,0 +1,522 @@
+	.request = sdhc_litex_request,
+	.set_io = sdhc_litex_set_io,
+	.get_card_present = sdhc_litex_get_card_present,
+	.card_busy = sdhc_litex_card_busy,
+	.get_host_props = sdhc_litex_get_host_props,
+};
+
+static int sdhc_litex_init(const struct device *dev)
+{
+	const struct sdhc_litex_config *dev_config = dev->config;
+
+	LOG_DBG("Initializing SDHC LiteX driver");
+
+	litex_write8(UINT8_MAX, dev_config->ev_pending_addr);
+	litex_write8(0, dev_config->ev_enable_addr);
+
+	dev_config->irq_config_func();
+
+	litex_write8(SDCARD_EV_BLOCK2MEM_DMA | SDCARD_EV_MEM2BLOCK_DMA, dev_config->ev_enable_addr);
+
+	litex_write8(SDCARD_PHY_SETTINGS_PHY_SPEED_1X, dev_config->phy_settings_addr);
+
+	return 0;
+}
+
+static void sdhc_litex_irq_handler(const struct device *dev)
+{
+	struct sdhc_litex_data *dev_data = dev->data;
+	const struct sdhc_litex_config *dev_config = dev->config;
+	uint8_t ev_enable = litex_read8(dev_config->ev_enable_addr);
+	uint8_t ev_pending = litex_read8(dev_config->ev_pending_addr) & ev_enable;
+
+	if (IS_BIT_SET(ev_pending, SDCARD_EV_CMD_DONE_BIT)) {
+		k_sem_give(&dev_data->cmd_done_sem);
+		litex_write8(ev_enable & ~SDCARD_EV_CMD_DONE, dev_config->ev_enable_addr);
+	}
+
+	if (IS_BIT_SET(ev_pending, SDCARD_EV_BLOCK2MEM_DMA_BIT) ||
+	    IS_BIT_SET(ev_pending, SDCARD_EV_MEM2BLOCK_DMA_BIT)) {
+		k_sem_give(&dev_data->dma_done_sem);
+	}
+
+	litex_write8(ev_pending, dev_config->ev_pending_addr);
+}
+
+#define DEFINE_SDHC_LITEX(n)                                                                       \
+	static void sdhc_litex_irq_config##n(void)                                                 \
+	{                                                                                          \
+		IRQ_CONNECT(DT_INST_IRQN(n), DT_INST_IRQ(n, priority), sdhc_litex_irq_handler,     \
+			    DEVICE_DT_INST_GET(n), 0);                                             \
+                                                                                                   \
+		irq_enable(DT_INST_IRQN(n));                                                       \
+	};                                                                                         \
+	static struct sdhc_litex_data sdhc_litex_data_##n = {                                      \
+		.lock = Z_MUTEX_INITIALIZER(sdhc_litex_data_##n.lock),                             \
+		.cmd_done_sem = Z_SEM_INITIALIZER(sdhc_litex_data_##n.cmd_done_sem, 0, 1),         \
+		.dma_done_sem = Z_SEM_INITIALIZER(sdhc_litex_data_##n.dma_done_sem, 0, 1),         \
+	};                                                                                         \
+	static const struct sdhc_litex_config sdhc_litex_config_##n = {                            \
+		.irq_config_func = sdhc_litex_irq_config##n,                                       \
+		.bus_width = (enum sdhc_bus_width)DT_INST_PROP(n, bus_width),                      \
+		.phy_card_detect_addr = DT_INST_REG_ADDR_BY_NAME(n, phy_card_detect),              \
```

Comment:
Do all these register offsets differ between each version of the IP? Or could we encode one base address?